File: android_mobile/src/main/java/com/alexstyl/specialdates/people/PeopleViewHolder.java
package com.alexstyl.specialdates.people;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.alexstyl.specialdates.images.ImageLoader;
import com.alexstyl.specialdates.ui.widget.ColorImageView;

class PeopleViewHolder extends RecyclerView.ViewHolder {

    private final ColorImageView avatarView;
    private final TextView nameView;
    private final ImageLoader imageLoader;

    PeopleViewHolder(View rowView, ImageLoader imageLoader, ColorImageView avatarView, TextView nameView) {
        super(rowView);
        this.imageLoader = imageLoader;
        this.avatarView = avatarView;
        this.nameView = nameView;
    }

    public void bind(final PersonViewModel viewModel, final PeopleViewHolderListener listener) {
        nameView.setText(viewModel.getPersonName());
        avatarView.setCircleColorVariant((int) viewModel.getPersonId());
        avatarView.setLetter(viewModel.getPersonName(), true);
        imageLoader.load(viewModel.getAvatarURI())
                .asCircle()
                .into(avatarView.getImageView());

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onPersonClicked(viewModel.getContact(), avatarView);
            }
        });
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/people/PeopleViewHolder.kt
package com.alexstyl.specialdates.people

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import com.alexstyl.specialdates.images.ImageLoader
import com.alexstyl.specialdates.ui.widget.ColorImageView

internal class PeopleViewHolder(rowView: View,
                                private val imageLoader: ImageLoader,
                                val avatarView: ColorImageView,
                                private val nameView: TextView) : RecyclerView.ViewHolder(rowView) {

    fun bind(viewModel: PersonViewModel, listener: PeopleViewHolderListener) {
        nameView.text = viewModel.personName
        avatarView.setCircleColorVariant(viewModel.personId.toInt())
        avatarView.setLetter(viewModel.personName, true)
        imageLoader.load(viewModel.avatarURI)
                .asCircle()
                .into(avatarView.imageView)

        itemView.setOnClickListener { listener.onPersonClicked(viewModel.contact, adapterPosition) }
    }
}
